test/integration/builder/triggers.js

```diff
@@ -0,0 +1,202 @@
+                    CREATE TRIGGER [${triggerName}] ON [${secondaryTable}]
+                    AFTER INSERT
+                    AS 
+                    BEGIN
+                        SET NOCOUNT ON;
+
+                        BEGIN
+                            update pt
+                            set pt.${primaryLink} = i.id
+                            from Inserted as i
+                            inner join ${primaryTable} as pt
+                                on pt.id = i.${secondaryLink}
+                        END
+                    END
+                `);
+            });
+
+            it('#4152 Should allow returns with inserts on tables with triggers', async function () {
+                if (knex.client.driverName !== 'mssql') {
+                    return true;
+                }
+
+                let reachedEnd = false;
+
+                await knex.transaction(async function () {
+                    let insertResults;
+
+                    async function insertWithReturn() {
+                        const insertPrimary = {
+                            data: "Testing Data"
+                        };
+
+                        const insertSecondary = {
+                            data: "Test Linking"
+                        }
+
+                        const primaryId = await knex(primaryTable).insert([insertPrimary], ["id"]);
+                        insertSecondary[secondaryLink] = primaryId[0];
+
+                        // Test retrieve with trigger
+                        insertResults = (await knex(secondaryTable).insert([insertSecondary], ["id"]))[0];
+                    }
+
+                    await insertWithReturn();
+
+                    expect(Number.parseInt(insertResults)).to.be.finite;
+
+                    reachedEnd = true;
+                });
+
+                expect(reachedEnd).to.be.true;
+            });
+        });
+
+        describe("Deletes with Triggers", function () {
+            // Trigger Name
+            const triggerName = 'tr_test_delete_with_trigger';
+
+            before(async function () {
+                if (knex.client.driverName !== 'mssql') {
+                    return;
+                }
+
+                await knex.raw(`
+                    CREATE TRIGGER [dbo].[${triggerName}] ON [dbo].[${secondaryTable}]
+                    AFTER DELETE
+                    AS 
+                    BEGIN
+                        SET NOCOUNT ON;
+
+                        BEGIN
+                            delete ${primaryTable}
+                            from ${primaryTable} as pt
+                                inner join deleted as d
+                                    on pt.id = d.${secondaryLink}
+                        END
+                    END
+                `);
+            })
+            it('#4152 Should allow returns with deletes on tables with triggers', async function () {
+                if (knex.client.driverName !== 'mssql') {
```

Comment:
if `this.skip` is executed in `before`, it shouldn't get here, so it shouldn't be necessary.